LibreNMS/Alert/Transport/Snmptrap.php

```diff
@@ -0,0 +1,173 @@
+
+/**
+ ** SNMP Trap Transport
+ **
+ ** @author Cloud Delivery Team <ttc-cloud-delivery@millicom.com>
+ ** @copyright 2022 Cloud Delivery Team, Tigo Technology Center
+ ** @license GPL
+ **/
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Config;
+
+class Snmptrap extends Transport
+{
+    /**
+     * @param  array  $obj
+     * @param  array  $opts
+     */
+    public function deliverAlert($obj, $opts)
+    {
+        $host = $this->config['snmptrap-destination-host'];
+        if (! empty($this->config['snmptrap-destination-port'])) {
+            $port = $this->config['snmptrap-destination-port'];
+        } else {
+            /* Default SNMP trap port */
+            $port = '162';
+        }
+        $transport = $this->config['snmptrap-transport'];
+        $trapdefinition = $this->config['snmptrap-definition'];
+        $pdu = $this->config['snmptrap-pdu'];
+        $community = $this->config['snmptrap-community'];
+        $binary = implode([Config::get('snmptrap')]);
+        $mibdir = $this->config['mib-dir'];
+
+        /**
+         * @param  array  $obj
+         */
+        return $this->contactSnmptrap($binary, $mibdir, $transport, $host, $port, $community, $trapdefinition, $pdu, $obj);
+    }
+
+    /**
+     * Returns if the call was successful
+     *
+     * @return bool
+     */
+    private function contactSnmptrap(string $binary, string $mibdir, string $transport, string $host, string $port, string $community, string $trapdefinition, string $pdu, array $obj)
+    {
+        $binary_opts = '';
+        switch ($pdu) {
+            case 'TRAPv2':
+                 $binary_opts = '-v 2c';
+                break;
+            case 'INFORM':
+                $binary_opts = '-v 2c -Ci';
+                break;
+            default:
+                echo 'This should not happen!!!';
+                break;
+        }
+
+        $msgsingle = preg_replace('~\R~', ' ', $obj['msg']);
+
+        putenv('SNMP_PERSISTENT_FILE=/tmp/snmpapp.conf.$USER');
+
+        //exec('/usr/bin/echo'
+        exec($binary
+            . ' ' . $binary_opts
+            . ' ' . '-M +' . $mibdir
+            . ' -c ' . $community
+            . ' ' . $transport . ':' . $host . ':' . $port
+            . ' ' . '\"\"' . ' ' . $trapdefinition . ' '
+            . $msgsingle, $output, $cmdexitcode
+        );
+
+        //var_dump($output);
+        if ($cmdexitcode == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static function configTemplate()
+    {
+        return [
+            'config' => [
+                [
+                    'title' => 'Destination host',
+                    'name' => 'snmptrap-destination-host',
+                    'descr' => 'Hostname or IP of the host that will receive the trap.',
+                    'type' => 'text',
+                ],
+                [
+                    'title' => 'Destination port',
```

Comment:
I think most of these settings should have a default value. (Either here or in the backend when they are accessed)

Reply:
It's already considered for port, it should be defined in a different way?. You see MIB::event as something that should have a default?, something else?

Reply:
In the webui, like this:

https://github.com/librenms/librenms/blob/master/LibreNMS/Alert/Transport/Mail.php#L67

Reply:
testing with 'default', but I don't see that default presented in the UI. Also, should I remove the 'required' portion in the validation section?

Reply:
@ciroiriarte yeah, I suspect default was only coded for checkboxes... just need to add it for input too.  If you feel up to it, you can add that, otherwise, it might take me a bit to get to it.  Changes will mostly be here: https://github.com/librenms/librenms/blob/master/includes/html/modal/edit_alert_transport.inc.php

Reply:
Took a quick glance. To be honest I wouldn't know where to start.